pkg/vms/actions.es6

```diff
@@ -0,0 +1,179 @@
+import cockpit from 'base1/cockpit';
+import Libvirt from 'vms/libvirt';
+import { getRefreshInterval } from 'vms/selectors';
+import { VMS_CONFIG } from "vms/config";
+
+/**
+ * All actions dispatchableby in the application
+ */
+
+// --- Provider actions -----------------------------------------
+export function getAllVms () {
+  return virt('GET_ALL_VMS');
+}
+
+export function getVm (lookupId) {
+  return virt('GET_VM', { lookupId }); // provider-specific (i.e. libvirt uses vm_name)
+}
+
+export function shutdownVm (name) {
+  return virt('SHUTDOWN_VM', { name });
+}
+
+export function forceVmOff (name) {
+  return virt('FORCEOFF_VM', { name });
+}
+
+export function rebootVm (name) {
+  return virt('REBOOT_VM', { name });
+}
+
+export function forceRebootVm (name) {
+  return virt('FORCEREBOOT_VM', { name });
+}
+
+export function startVm (name) {
+  return virt('START_VM', { name });
+}
+
+/**
+ * Helper for dispatching virt provider methods.
+ *
+ * Lazily initializes the virt provider and dispatches given method on it.
+ */
+function virt (method, action) {
+  return (dispatch, getState) => getVirtProvider({ dispatch, getState }).then(provider => {
+    if (method in provider) {
+      console.log(`Calling ${provider.name}.${method}(${JSON.stringify(action)})`);
+      return dispatch(provider[method](action));
+    } else {
+      console.warn(`method: '${method}' is not supported by provider: '${provider.name}'`);
+    }
+  }).catch(err => {
+    console.error('could not detect any virt provider');
+  })
+}
+
+function getVirtProvider(store) {
+  const state = store.getState();
+  if (state.config.provider) {
+    return cockpit.resolve(state.config.provider);
+  } else {
+    const deferred = cockpit.defer();
+    console.log('Discovering provider');
```

Comment:
Does this display *either* libvirt or another provider? In which case what do you think about this being a part of build dependencies, resulting in separate conflicting cockpit-libvirt and cockpit-anotherprovider subpackages?